crates/ra_ide_api/src/syntax_highlighting.rs

```diff
@@ -61,7 +69,10 @@ pub(crate) fn highlight(db: &RootDatabase, file_id: FileId) -> Vec<HighlightedRa
             STRING | RAW_STRING | RAW_BYTE_STRING | BYTE_STRING => "string",
             ATTR => "attribute",
             NAME_REF => {
-                if let Some(name_ref) = node.as_node().and_then(ast::NameRef::cast) {
+                // FIXME: revisit this after #1340
+                if is_prim_type(&node) {
```

Comment:
It's better to use `node.text().as_str()` to avoid allocation

Reply:
That is, in the branch where we've already casted to ast::NameRef: `.text` is method on `NameRef`

Reply:
Can we keep ignoring `SyntaxToken`s`? I'm still not sure what those are.

Reply:
`SyntaxToken` is the leaf element of  a syntax tree. Basically, a string with a tag.

`SyntaxNode` is a composite element of a syntax tree, which has other Nodes or Tokens as children

`SyntaxElement` is either of the two

Reply:
I don't see in `SyntaxText` any `as_str()` method that doesn't allocate.

Reply:
https://github.com/rust-analyzer/rust-analyzer/blob/cb21a219987eac0f8dc08f0c858e8a7154fccd04/crates/ra_syntax/src/ast/extensions.rs#L22

Reply:
But never mind, `==` probably works.